### PR TITLE
More reliable dfx-stop

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
`dfx stop` _should_ send the `dfx start` process a `SIGTERM` signal.  But sometimes it doesn't.

# Changes
- Use pgrep to find `dfx start` (ar `dfx --some-flags start`) and send the signal.

### CODE REVIEW

File: bin/dfx-network-stop [hunk 1]
```diff
@@ -14,6 +14,8 @@ export DFX_NETWORK
 if [[ "$DFX_NETWORK" == "local" ]]; then
   : "Ask for a clean shutdown"
   dfx stop || true
+  : "Also send the equivalent of ctrl-c, which _should_ be equivalent, as dfx stop often fails"
+  pgrep --full '^dfx\>.*\<start\>' | xargs --no-run-if-empty kill
```
[the remark this answers is not in the document]
Comment (bitdivine) [reply bin/dfx-network-stop:18]: Here is a better idea:  Don't use dfx stop at all.  Either `dfx start` is global or it is not.  If global, it doesn't make sense to have a `dfx stop` command that isn't global.

[the remark this answers is not in the document]
Comment (dskloetd) [reply bin/dfx-network-stop]: Well, that's an idea for the sdk team. Can you check with them what are (and might in the future be) the differences between `dfx stop` and `^C`? If we're going to decide that the cleanup that `dfx stop` does that `^C` doesn't, we should at least know what the difference is.

[the remark this answers is not in the document]
Comment (bitdivine) [reply bin/dfx-network-stop]: This entire repo is ideas for the sdk team. Test ideas or improvements, then push them upstream when they have proven themselves.

**Comment by dskloetd** [comment]
As I said, I've never seen `dfx stop` not work and I have seen `^C` fail to do something that `dfx stop` does, so from my perspective this change does not look like an improvement.

**Pull request merged**